sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py

```diff
@@ -551,56 +748,144 @@ def _extract_data_cb(self, get_next_return):
+    """Enable/disable 'NotContentIndexed' attribute for DIRECTORY."""
+    no_scrub_data: bool = False
+    """Enable/disable 'NoScrubData' attribute for DIRECTORY."""
+
+    def __init__(
+        self, read_only: bool = False,
+        hidden: bool = False,
+        system: bool = False,
+        none: bool = False,
+        directory: bool = False,
+        archive: bool = False,
+        temporary: bool = False,
+        offline: bool = False,
+        not_content_indexed: bool = False,
+        no_scrub_data: bool = False
+    ) -> None:
+        self.read_only = read_only
+        self.hidden = hidden
+        self.system = system
+        self.none = none
+        self.directory = directory
+        self.archive = archive
+        self.temporary = temporary
+        self.offline = offline
+        self.not_content_indexed = not_content_indexed
+        self.no_scrub_data = no_scrub_data
+        self._str = (('ReadOnly|' if self.read_only else '') +
+                               ('Hidden|' if self.hidden else '') +
+                               ('System|' if self.system else '') +
+                               ('None|' if self.none else '') +
+                               ('Directory|' if self.directory else '') +
+                               ('Archive|' if self.archive else '') +
+                               ('Temporary|' if self.temporary else '') +
+                               ('Offline|' if self.offline else '') +
+                               ('NotContentIndexed|' if self.not_content_indexed else '') +
+                               ('NoScrubData|' if self.no_scrub_data else ''))
+
+    def __str__(self):
+        concatenated_params = self._str
+        return concatenated_params.strip('|')
+
+    @classmethod
+    def from_string(cls, string: str) -> "NTFSAttributes":
+        """Create a NTFSAttributes from a string.
+
+        To specify permissions you can pass in a string with the
+        desired permissions, e.g. "ReadOnly|Hidden|System"
+
+        :param str string: The string which dictates the permissions.
+        :return: A NTFSAttributes object
+        :rtype: ~azure.storage.fileshare.NTFSAttributes
+        """
+        read_only = "ReadOnly" in string
+        hidden = "Hidden" in string
+        system = "System" in string
+        none = "None" in string
+        directory = "Directory" in string
+        archive = "Archive" in string
+        temporary = "Temporary" in string
+        offline = "Offline" in string
+        not_content_indexed = "NotContentIndexed" in string
+        no_scrub_data = "NoScrubData" in string
+
+        parsed = cls(read_only, hidden, system, none, directory, archive, temporary, offline, not_content_indexed,
+                     no_scrub_data)
+        parsed._str = string  # pylint: disable = protected-access
+        return parsed
+
+
+class DirectoryProperties(DictMixin):
+    """Directory's properties class."""
+
+    name: Optional[str] = None
```

Comment:
The `name` is not `Optional` because the generated code will set it down the line. There are not really cases where it's not set, but here the starting type of the instance variable `self.name` is technically `None`. How about we make name `str` since that's its true type, then put a `# type: ignore [assignment]`?

Reply:
It feels janky but I think that's how the codebase is structured.